internal/topo/context/transform.go

```diff
@@ -20,13 +20,27 @@
 )
 
 const TransKey = "$$trans"
+const SelectKey = "$$select"
+const SelectFields = "$$selectFields"
 
 // TransformOutput Lazy transform output to bytes
 func (c *DefaultContext) TransformOutput(data interface{}) ([]byte, bool, error) {
 	v := c.Value(TransKey)
 	f, ok := v.(transform.TransFunc)
-	if ok {
-		return f(data)
+	if !ok {
+		return nil, false, fmt.Errorf("no transform configured")
 	}
-	return nil, false, fmt.Errorf("no transform configured")
+
+	bytes, ok, err := f(data)
+	if err != nil || ok || c.Value(SelectFields) == nil {
+		return bytes, ok, err
+	}
+
+	fields := c.Value(SelectFields).([]string)
+	s := c.Value(SelectKey)
+	sf, ok := s.(transform.SelectFunc)
+	if !ok {
+		return bytes, false, fmt.Errorf("no select configured")
```

Comment:
What's the relationship between transform(template, format) and fields? We need to clearly define and document it.